cvat/apps/dataset_manager/views.py

```diff
@@ -127,14 +176,55 @@ def export_project_annotations(project_id, dst_format=None, server_url=None):
     return export(dst_format, project_id=project_id, server_url=server_url, save_images=False)
 
 
-def clear_export_cache(file_path, file_ctime, logger):
+def clear_export_cache(file_path: str, file_ctime: float, logger: logging.Logger) -> None:
     try:
-        if osp.exists(file_path) and osp.getctime(file_path) == file_ctime:
+        with get_dataset_cache_lock(
+            file_path,
+            block=True,
+            acquire_timeout=EXPORT_CACHE_LOCK_TIMEOUT,
+            ttl=rq.get_current_job().timeout,
+        ):
+            if 'job' in file_path:
+                instance_type = 'job'
+            elif 'task' in file_path:
+                instance_type = 'task'
+            elif 'project' in file_path:
+                instance_type = 'project'
+            else:
+                assert False
+
+            cache_ttl = get_export_cache_ttl(instance_type)
+
+            instance_timestamp = None
+            with suppress(AttributeError): # backward compatibility
+                instance_timestamp = get_file_instance_timestamp(file_path)
+            if instance_timestamp and instance_timestamp != file_ctime:
+                logger.info("Export cache file '{}' has changed, skipping".format(file_path))
+                return
+
+            if timezone.now().timestamp() <= osp.getmtime(file_path) + cache_ttl:
+                # Need to retry later, the export is in use
+                rq_job = rq.get_current_job()
+                rq_job.retries_left = 1
+                rq_job.retry_intervals = [cache_ttl]
+                rq_job.retry(
```

Comment:
IMO, instead of retrying, it would be clearer to enqueue another job with the same parameters at the desired time.

Reply:
Maybe. The difference is that if the job name is preserved, the results from multiple calls can be displayed as a list in Django admin. I found it more representative.

Reply:
Okay, that's a fair point.